src/controllers/things_controller.js

```diff
@@ -26,10 +26,39 @@ const ThingsController = PromiseRouter();
  * Get a list of Things.
  */
 ThingsController.get('/', function(request, response) {
-  Things.getThingDescriptions(request.get('Host'), request.secure)
-    .then(function(things) {
-      response.status(200).json(things);
-    });
+  if (request.jwt.payload.role !== Constants.USER_TOKEN) {
+    if (!request.jwt.payload.scope) {
+      response.status(400).send('Token must contain scope');
+    } else {
+      const scope = request.jwt.payload.scope;
+      if (scope.indexOf(' ') === -1 && scope.indexOf('/') == 0 &&
+        scope.split('/').length == 2) {
```

Comment:
This has a small bug because authorization codes have scope `/oauth:readwrite`. It would be better to make sure that scope.split(':')[0] is equal to Constants.THINGS_PATH.